components/canvas/gl_context.rs

```diff
@@ -141,6 +160,8 @@ impl GLContextFactory {
 pub enum GLContextWrapper {
     Native(GLContext<NativeGLContext>),
     OSMesa(GLContext<OSMesaContext>),
+    #[cfg(target_os = "macos")]
+    NativeWithIOSurface(GLContext<NativeGLContext>),
```

Comment:
My overall goal in my suggestions is to remove this new enum variant and move the platform-specific behaviour either into rust-offscreen-rendering-context by adding abstractions like TextureBacking, or use match statements on TextureBacking values in this code to choose which behaviour to use.